### ISSUE TITLE
Is this a bug in dqn.compile() ?

### ISSUE THREAD

**Issue by AxKo** [keras-rl/keras-rl#215]
Hi everybody,
I noticed that in dqn.compile() the optimizer to compile self.model is always 'sgd'.
I understand that for self.target_model the optimizer doesn't matter, but I don't understand why this should be the case for self.model.

`        # We never train the target model, hence we can set the optimizer and loss arbitrarily.
        self.target_model = clone_model(self.model, self.custom_model_objects)
        self.target_model.compile(optimizer='sgd', loss='mse')    # I understand this                      
        self.model.compile(optimizer='sgd', loss='mse')               # But not this
`

However, I did actually change the last line and did not see any difference when running one of my models. So I guess the question is really why does the optimizer that is passed to self.model.compile() not have any influence on the training ??

Thanx,
    Axel

**Comment by ryou90** [comment]
Same Question a few days ago #207
